src/components/media-views.js

```diff
@@ -586,11 +592,19 @@ AFRAME.registerComponent("media-video", {
     return new Promise(async (resolve, reject) => {
       const videoEl = await createVideoEl();
 
-      const texture = new THREE.VideoTexture(videoEl);
-      texture.minFilter = THREE.LinearFilter;
-      texture.encoding = THREE.sRGBEncoding;
-      const isReady = () =>
-        (texture.image.videoHeight || texture.image.height) && (texture.image.videoWidth || texture.image.width);
+      let texture, isReady;
+      if (contentType === "audio/mpeg") {
+        // We want to treat audio almost exactly like video, so we mock a video texture with an image property.
+        texture = new THREE.Texture();
```

Comment:
We only create this texture so we can pull the video element off of it, and then later assign it to the map of the material. In the audio case we skip the second part, so it seems we could skip this whole thing by pulling `createVideoEl()` out into the calling code, and then only calling `createVideoTexture` for videos (passing in the videoEl)

Reply:
Eh. I see what you mean but it seems like it could get hairy since we use texture as a container for things everywhere. It also contains this `hls` property and I was thinking it ought to work with audio as well in the future. So I'm going to punt on the restructuring for now.